Source/DafnyCore/Rewriters/AutoRevealFunctionDependencies.cs

```diff
@@ -0,0 +1,309 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Numerics;
+
+namespace Microsoft.Dafny;
+
+/// <summary>
+/// When Dafny is called with `--default-function-opacity autoRevealDependencies`, this rewriter computes
+/// all transitive functional dependencies for each callable, and inserts (in-memory, on the AST) reveal stmts for each such dependency
+/// at the top of the callable body, and also in a precondition.
```

Comment:
```suggestion
/// at the top of the callable body, and also in any preconditions.
```

Reply:
So actually, all the reveal stmts are collected in a single requires stmt (like `requires reveal f1(); reveal f2(); reveal f3(); true`), and inserted at the beginning, so I believe the original is more accurate.

Reply:
Maybe "at the top of all clauses"?